### ISSUE TITLE
Hide directive syntax when not editing

### ISSUE THREAD

**Issue by yorrd** [silverbulletmd/silverbullet#144]
As a user, I don't want to see the directive syntax when not editing it directly. If the cursor is not within the result of the query, those lines could be hidden. 

Example: 
![image](https://user-images.githubusercontent.com/1191486/205496422-602c6e6a-b25e-4e3c-a778-1e8d4d43a9df.png)

Here I'd like to just see the resulting link as generated by the directive.

**Comment by zefhemel** [comment]
Yeah, I played with this, but the UX issue is: should this be 100% hidden? When should the directive appear? At some point I just showed the # and / little widget, but not the directive code, but that just results in a lot of whitespace. Suggestions?

**Comment by yorrd** [comment]
what about this: when the user's cursor enters in the contained text, the enclosing syntax is shown? Just like in the code blocks, the first line is also hidden when not within the code block.

**Comment by zefhemel** [comment]
Right, that's worth trying.

**Comment by zefhemel** [comment]
Reopening because I want to add some sort of visual indication, perhaps a border-left for directive bodies. To make it somewhat visual where directives are at work.
